apps/coursition/src/app/api/auth/[...nextauth]/auth-options.ts

```diff
@@ -0,0 +1,54 @@
+import { PrismaAdapter } from '@auth/prisma-adapter'
+import { prisma } from 'apps/coursition/prisma/prismaClient'
+import bcrypt from 'bcryptjs'
+import { Effect, Either, Struct } from 'effect'
+import type { NextAuthOptions } from 'next-auth'
+import type { Adapter } from 'next-auth/adapters'
+import CredentialsProvider from 'next-auth/providers/credentials'
+import GoogleProvider from 'next-auth/providers/google'
+
+const handleAuthorize = (dto: { email: string; password: string }) =>
+  Effect.gen(function* () {
+    const result = yield* Effect.tryPromise({
+      try: async () => await prisma.user.findFirstOrThrow({ where: { email: dto.email } }),
+      catch: () => Effect.fail('user not found'),
```

Comment:
**blocker:** We don't handle invalid password and failing effect like this is not correct way. We should use Expected Error (mentioned in review above).